### ISSUE TITLE
hlint incorrectly suggests renaming shadowed "map" when generalise group is on

### ISSUE THREAD

**Issue by isovector** [ndmitchell/hlint#1572]
Given 

```yaml
- group: {name: generalise, enabled: true}
```

hlint will suggest the following illegal transformation:

```diff
- test map = map
+ test map = fmap
```

The fix is to fully qualify the following:

https://github.com/ndmitchell/hlint/blob/8bd116cc7a3fcabf309cb9c1b9aa35016c190b09/data/hlint.yaml#L1125

to 

```yaml
    - warn: {lhs: Data.List.map, rhs: fmap}
```
